Added the Rating model

AlonShakaroff opened the pull request:
- Added the rating table and functions
- Added functions to resume that refer to the ratings of the function
- Added tests to the Rating, and to the Resume functions.

fix #55 
depends on #50

OmerCS8 reviewed (dismissed):
LGTM 🥇

TomerNewmanPrograms reviewed (approved):
lgtm

MatanP12 reviewed (approved):
LGTM

OmerCS8 reviewed (approved):
LGTM

MatanP12 commented on posts/tests.py:
```diff
@@ -1,3 +1,201 @@
+        assert isinstance(new_resume, Post)
+        assert len(Post.objects.all()) == 1
+        post = Post.objects.all().first()
+        assert hasattr(post, 'resume') is True
```
change here as well

MatanP12 commented on posts/tests.py:
```diff
@@ -1,3 +1,201 @@
+        assert len(Post.objects.all()) == 1
+        post = Post.objects.all().first()
+        assert hasattr(post, 'resume') is True
+        assert hasattr(post, 'poll') is False
```
same

natifridman commented on posts/models.py:
```diff
@@ -1,13 +1,91 @@
+    def __str__(self):
+        return f"Resume {self.post_id} by {self.author}"
+
+    def get_average_rating(self):
+        ratings = self.rating_set.all()
+        rating_sum = 0
+        for rating in ratings:
+            rating_sum += rating.get_rating_average()
+        return rating_sum / len(self.rating_set.all())
+
+    def get_design_rating(self):
+        ratings = self.rating_set.all()
+        rating_sum = 0
+        for rating in ratings:
+            rating_sum += rating.design_rating
+        return rating_sum / len(self.rating_set.all())
+
+    def get_skill_relevance_rating(self):
+        ratings = self.rating_set.all()
+        rating_sum = 0
+        for rating in ratings:
+            rating_sum += rating.skill_relevance_rating
+        return rating_sum / len(self.rating_set.all())
+
+    def get_grammar_rating(self):
+        ratings = self.rating_set.all()
+        rating_sum = 0
+        for rating in ratings:
+            rating_sum += rating.grammar_rating
+        return rating_sum / len(self.rating_set.all())
+
+    def get_conciseness_rating(self):
+        ratings = self.rating_set.all()
+        rating_sum = 0
+        for rating in ratings:
+            rating_sum += rating.conciseness_rating
+        return rating_sum / len(self.rating_set.all())
```
I think you can use these methods as [class method](https://stackabuse.com/pythons-classmethod-and-staticmethod-explained/), so you can use [cls.objects.filter](https://github.com/redhat-beyond/falcon/blob/c719a8365b2f146224f763478d978acad5c39c02/tasks/models.py#L37)

AlonShakaroff replied:
Ok, I'll check that out, thanks!

AlonShakaroff replied:
If I use these classes as class methods I can't use the object it self to get it's rating_set, so I don't see how am I supposed to get the rating average with the method as class method? @natifridman

natifridman replied:
I thought it was in the Rating model, you can ignore this comment if you keep it in the Resume model

MatanP12 reviewed (changes requested):
Please delete the .idea folder from the PR

AlonShakaroff commented:
> Please delete the .idea folder from the PR

Oops, my bad, fixed it 👍

OmerCS8 reviewed (dismissed):
LGTM

natifridman commented on posts/models.py:
```diff
@@ -21,4 +23,70 @@ class Resume(Post):
+        rating_sum = 0
+        for rating in ratings:
+            rating_sum += rating.conciseness_rating
+        return rating_sum / len(self.rating_set.all())
```
same

AlonShakaroff commented on posts/models.py:
```diff
@@ -21,4 +23,70 @@ class Resume(Post):
+        rating_sum = 0
+        for rating in ratings:
+            rating_sum += rating.get_rating_average()
+        return rating_sum / len(self.rating_set.all())
```
Yeah, you're right, changed that in all the methods 👍 
@natifridman

OmerCS8 reviewed (dismissed):
🚀

MatanP12 reviewed (approved):
🔥

natifridman commented:
@AlonShakaroff Maybe we can add some tests for the average methods?

AlonShakaroff commented:
> @AlonShakaroff Maybe we can add some tests for the average methods?

I wrote them already and didn't upload them by mistake, I'll upload them soon.

AlonShakaroff commented:
> @AlonShakaroff Maybe we can add some tests for the average methods?

Added the tests for the methods, and also fixed potential division by zero bugs.
@natifridman

yulisuliman reviewed (approved):
Looks good!

natifridman commented on posts/models.py:
```diff
@@ -21,4 +23,92 @@ class Resume(Post):
+        ratings = self.rating_set.all()
+        rating_sum = 0
+        ratings_len = len(ratings)
+
```
Please remove the [blank line](https://peps.python.org/pep-0008/#blank-lines)

natifridman commented on posts/models.py:
```diff
@@ -21,4 +23,92 @@ class Resume(Post):
+
+        for rating in ratings:
+            rating_sum += rating.get_rating_average()
+
```
Please remove the [blank line](https://peps.python.org/pep-0008/#blank-lines)

natifridman commented on posts/tests.py:
```diff
@@ -75,6 +118,24 @@ def test_resume_deletion_after_author_deletion(self, persist_resume):
+    def test_resume_average_functions(self, persist_resume):
+        current_grade = 5
+        current_username = "test_rater"
+
```
Please remove the [blank line](https://peps.python.org/pep-0008/#blank-lines)

natifridman commented on posts/tests.py:
```diff
@@ -75,6 +118,24 @@ def test_resume_deletion_after_author_deletion(self, persist_resume):
+            current_rater = new_user_with_name_and_email(current_username, current_email)
+            new_rating_with_input(resume=persist_resume, author=current_rater, grade=current_grade)
+            current_grade -= 2
+
```
Please remove the [blank line](https://peps.python.org/pep-0008/#blank-lines)

natifridman reviewed (changes requested):
Please follow [pep8 for blank line style](https://peps.python.org/pep-0008/#blank-lines)

AlonShakaroff commented:
> Please follow [pep8 for blank line style](https://peps.python.org/pep-0008/#blank-lines)

Removed the blank lines. @natifridman

MatanP12 reviewed (approved):
LGTM
